### ISSUE TITLE
Performance drop in Test Case 1: SPDK NVMe-oF TCP Target I/O core scaling

### ISSUE THREAD

**Issue by wawryk** [spdk/spdk#2159]
# Bug report

## Expected Behavior
Have constant level of performance results

## Current Behavior
Have IOPS drop.

## Possible Solution
When doing bisect between v21.04 and v21.07 have found possible commit:
f5ba8a5ef50485f182c07e1b52fccedc17cbd72d IOPS results: 
```
441615.178
376561.136 
438604.490 
420837.525 
448301.803
```

last good commit, what gives me good IOPS results:
df01076f70f6f61255e4151ab32216d10bc48e88: 
```
1123987.183 
1158506.581 
1065133.060 
1010948.313 
1281015.008
```

## Steps to Reproduce

config.json:
```
{
  "general": {
    "username": "sys_sgci",
    "password": "espidikej",
    "transport": "tcp"
  },
  "target": {
    "nic_ips": ["20.0.0.1", "20.0.1.1", "10.0.0.1", "10.0.1.1"],
    "mode": "spdk",
    "use_null_block": false,
    "nvmet_dir": "/home/sys_sgci/nvmetcli",
    "core_mask": "[0-1,20-21]",
    "zcopy_settings": true,
    "sar_settings": [true, 10, 1, 90]
  },
  "initiator1": {
    "ip": "10.211.11.74",
    "nic_ips": ["20.0.0.2", "20.0.1.2"],
    "target_nic_ips": ["20.0.0.1", "20.0.1.1"],
    "fio_dir": "/usr/src/fio",
    "mode": "kernel",
    "extra_params": "",
    "num_cores": 4
  },
  "initiator2": {
    "ip": "10.211.11.73",
    "nic_ips": ["10.0.0.2", "10.0.1.2"],
    "target_nic_ips": ["10.0.0.1", "10.0.1.1"],
    "fio_dir": "/usr/src/fio",
    "mode": "kernel",
    "extra_params": "",
    "num_cores": 4
  },
  "fio": {
    "bs": ["4k"],
    "qd": [64],
    "rw": ["randwrite"],
    "rwmixread": 70,
    "num_jobs": 1,
    "run_time": 120,
    "ramp_time": 10,
    "run_num": 1
  }
}
```
1. Run run_nvmf.py script located in spdk/scripts/perf/nvmf/run_nvmf.py

**Comment by jimharris** [comment]
Hi @wawryk,

A couple of questions:
* Do you only see this for 70/30 RW mix test or does it also happen on 100% read and 100% write tests?
* Do you see the problem if only 1 core running on the target, or does it only happen with 4 or more?

I'm looking at the patch in question - it changed when we read the CAP register - it's now done after we've connected the admin queue rather than when the ctrlr is constructed.  My first thought was it impacts things like default io_queue_size and io_queue_requests, but code inspection and some nvme pcie tests on my system all check out OK.  Still investigating...

-Jim

**Comment by changpe1** [comment]
[Bug Scrub] @wawryk we can assign this issue to other after getting some data.

**Comment by wawryk** [comment]
@jimharris 
I see this only on 100% writes, sorry for inconvenience in previous message. 
For explanation `"rwmixread": 70,` parameter is ignored by script, when rw is set to randread or randrwrite, and then we have 100% reads or writes.
 I see these problem for cores from 1 to 16 (on attached picture You can see diffs between v21.07 and v21.04):
![image](https://user-images.githubusercontent.com/15612188/133558236-0d05cd92-2e49-45ae-aadd-0100d98221dc.png)
I choose 4 cores, because in our perf test, there was the highest diff.

**Comment by jimharris** [comment]
@karlatec @xjjx Can this issue be closed?  Looking at 22.01 NVMe/TCP performance report, the results look similar to what was reported in the v21.04 (good) columns above.

**Comment by karlatec** [comment]
I think so. Too bad that this one wasn't updated with description on how was this resolved.
@wawryk can this be closed? If possible please add a comment what was the root cause and how it was solved.
